src/transformers/trainer.py

```diff
@@ -2472,6 +2471,9 @@ def _inner_training_loop(
                             input_tokens = inputs[main_input_name].numel()
                             input_tokens = torch.tensor(input_tokens, device=self.args.device, dtype=torch.int64)
                             self.state.num_input_tokens_seen += self.accelerator.gather(input_tokens).cpu().item()
+                            self.log(
+                                speed_metrics("train_step", start_time, num_tokens=self.state.num_input_tokens_seen)
+                            )
```

Comment:
Maybe we can put that in `_maybe_log_save_evaluate`  function instead as it is called at the end of each batch

Reply:
Thanks for the hint, I was not aware of this function. Moved the logic now.